src/test/datascience/vscodeTestHelpers.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+'use strict';
+import { createDeferred } from '../../client/common/utils/async';
+
+// Basic shape that something needs to support to hook up to this
+interface IOnMessageListener {
+    addMessageListener(callback: (message: string, payload: any) => void): void;
+    removeMessageListener(callback: (message: string, payload: any) => void): void;
+}
+
+export type WaitForMessageOptions = {
+    /**
+     * Timeout for waiting for message.
+     * Defaults to 65_000ms.
+     *
+     * @type {number}
+     */
+    timeoutMs?: number;
+    /**
+     * Number of times the message should be received.
+     * Defaults to 1.
+     *
+     * @type {number}
+     */
+    numberOfTimes?: number;
+
+    // Optional check for the payload of the message
+    // will only return (or count) message if this returns true
+    // tslint:disable-next-line: no-any
+    withPayload?(payload: any): boolean;
+};
+
+// This class is for usage in .vscode test to hook up to webviews which expose the addMessageListener and removeMessageListener functions
+export class OnMessageListener {
+    private target: IOnMessageListener;
+    constructor(target: IOnMessageListener) {
+        this.target = target;
+    }
+
+    // For our target object wait for a specific message to come in from onMessage function
+    public async waitForMessage(message: string, options?: WaitForMessageOptions): Promise<void> {
```

Comment:
The waitForMessage implementation here is very close to the existing one in mountedWebView.ts. Figured that interface already existed and was doing the same basic thing so just reused it here with minor changes for not being part of a mountedWebView.